lib/node_modules/@stdlib/stats/incr/wmean/docs/types/index.d.ts

```diff
@@ -0,0 +1,52 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2019 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/// <reference types="@stdlib/types"/>
+
+/**
+* Returns an accumulator function which incrementally computes a weighted arithmetic mean.
+*
+* @returns accumulator function
+*
+* @example* var incrwmean = require( '@stdlib/stats/incr/wmean' );
+*
+* var accumulator = incrwmean();
+*
+* var mu = accumulator();
+* // returns null
+*
+* var mu = accumulator( 2.0, 1.0 );
+* // returns 2.0
+*
+* mu = accumulator( 2.0, 0.5 );
+* // returns 2.0
+*
+* mu = accumulator( 3.0, 1.5 );
+* // returns 2.5
+*
+* mu = accumulator();
+* // returns 2.5
+*/
+declare function incrwmean(): ( x: number, w: number ) => number | null;
```

Comment:
I think declaring a type definition for the return value is probably better here and then using that type definition for the return value. The reason being that, by creating the type definition, we are able to explicitly document the accumulator parameters and return values.